### PR TITLE
Required Actions (squashed relevant commits from pr 123)

### PR DESCRIPTION
**Pull request by tomrutsaert**
squashed relevant commits from pr https://github.com/mrparkers/terraform-provider-keycloak/pull/123

### CODE REVIEW

File: keycloak/required_action.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		return err
+	}
+
+	if requiredAction.DefaultAction && !requiredAction.Enabled {
+		return fmt.Errorf("validation error: a 'default' required action should be enabled, set 'defaultAction' to 'false' or set 'enabled' to 'true'")
+	}
+
+	if !serverInfo.providerInstalled("required-action", requiredAction.Alias) {
+		return fmt.Errorf("validation error: required action \"%s\" does not exist on the server, installed providers: %s", requiredAction.Alias, serverInfo.getInstalledProvidersNames("required-action"))
+	}
```
Comment (tomrutsaert) [inline keycloak/required_action.go]: This is the extra validation check for disabled default actions

File: provider/resource_keycloak_required_action.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	parts := strings.Split(d.Id(), "/")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid import. Supported import formats: {{realmId}}/{{alias}}")
+	}
+	d.Set("realm_id", parts[0])
+	d.Set("alias", parts[1])
+	d.SetId(fmt.Sprintf("%s/%s", parts[0], parts[1]))
+
+	return []*schema.ResourceData{d}, nil
+}
```
Comment (tomrutsaert) [inline provider/resource_keycloak_required_action.go]: @mrparkers Please check this import function.

Comment (mrparkers) [reply provider/resource_keycloak_required_action.go]: Looks perfect, nice job!

**Comment by mrparkers** [comment]
Running `make fmt` should help with the failing CI build.

**Comment by timbmg** [comment]
Is it possible to assign the required action to a user or group?

**Comment by timbmg** [comment]
I just did a little bit of digging, and from what I can understand fro the doc, is that keycloak would support creating/updating required actions for users.
The user resource takes a list of strings for required actions. See here: [https://www.keycloak.org/docs-api/10.0/rest-api/index.html#_userrepresentation](https://www.keycloak.org/docs-api/10.0/rest-api/index.html#_userrepresentation)

**Pull request merged**